### ISSUE TITLE
self.class_embed = nn.Linear(hidden_dim, num_classes)

### ISSUE THREAD

**Issue by Innary** [fundamentalvision/Deformable-DETR#152]
不好意思想问一下为什么
deformable-DETR中类别分类是
`self.class_embed = nn.Linear(hidden_dim, num_classes)`
num_classes=91
在这里是91类，而在标准的DETR中是92类（91类+1背景类）？
是做了什么映射操作么

**Comment by EMU1337X** [comment]
同问

**Comment by cdluminate** [comment]
原版DETR用的cross entropy loss, 因为COCO自身只有一些N/A类别，不带背景类（用于表示该prediction head所对应的物体什么都不是），所以增加背景类扩展为了92。
Deformable DETR用的focal loss (binary cross entropy)，这种情况下因为label可以是zero-hot vector, 不需要扩展到92类。

**Comment by Pujin823** [comment]
所以只要在one-hot编码中对应全0的都会被视为背景类别吗，因为coco只有80类，但最只是最高的类别索引为90，中间一些索引并不对应类别（比如索引0），这些经过one-hot之后都是全0的吧
